src/base64_encoder.cpp

```diff
@@ -0,0 +1,111 @@
+/**
+ * @file
+ * @brief     Implementation for base64encoder
+ * @author    Arsalan Anwari
+ * @license   See LICENSE
+ */
+#include "base64_encoder.hpp"
+
+Camera::Base64Encoder::Base64Encoder(std::string &filePath) : filePath(filePath) {
+}
+
+Camera::Base64Encoder::Base64Encoder() {
+}
+
+std::vector<unsigned char> &Camera::Base64Encoder::getByteStream() {
+    buffer.clear();
+    std::ifstream file(filePath, std::ios::binary);
+    std::istream_iterator<unsigned char> begin(file), end;
+    buffer.assign(begin, end);
+    return buffer;
+}
+
+void Camera::Base64Encoder::setFilePath(std::string &newFilePath) {
+    filePath = newFilePath;
+}
+
+std::string Camera::Base64Encoder::encode() {
+
+    std::vector<unsigned char> &data = getByteStream();
```

Comment:
This function is exactly the same as the other encode() function except for this line. Why not call that function after you get the data with this line.